CryptoLightNews/Networking/Service.swift

```diff
@@ -0,0 +1,53 @@
+//
+//  Service.swift
+//  CryptoLightNews
+//
+//  Created by Richard Price on 21/12/2022.
+//
+
+import Foundation
+
+class Serivce {
+    static let shared = Serivce()
+    typealias News = Result<SearchResults, CryptoNewsErrors>
+    let constants = Constants()
+    
+    func fetchNews(completion: @escaping (News) -> Void) {
+        let urlString = CryptoNewsAPI.getNews
+        fetchGenericJSONData(urlString: urlString.rawValue, completion: completion)
+    }
+    
+    func fetchGenericJSONData<T: Decodable>(urlString: String, completion: @escaping (Result<T, CryptoNewsErrors>) -> ()) {
```

Comment:
Maybe you should verify the URL before you call this function. And make it private.  
```suggestion
    private func fetchGenericJSONData<T: Decodable>(url: URL, completion: @escaping (Result<T, CryptoNewsErrors>) -> ()) {
```

Reply:
struggling with this one unable to get it to work with the suggestion